apps/ejabberd/src/mongoose_http_client.erl

```diff
@@ -70,72 +79,63 @@ post(Pool, Path, Headers, Query) ->
     make_request(Pool, Path, <<"POST">>, Headers, Query).
 
 %%------------------------------------------------------------------------------
-%% gen_mod callbacks
+%% exported for testing
 
--spec start(ejabberd:server(), list()) -> any().
-start(Host, Opts) ->
-    {ok, _SupProc} = start_supervisor(Host),
+start(Opts) ->
+    {ok, _SupProc} = start_supervisor(),
     EjdSupPid = whereis(ejabberd_sup),
     HeirOpt = case self() =:= EjdSupPid of
                   true -> [];
                   false -> [{heir, EjdSupPid, self()}] % for dynamic start
               end,
-    ets:new(tab_name(Host), [set, public, named_table, {keypos, 2},
-                             {read_concurrency, true} | HeirOpt]),
-    PoolSpec = gen_mod:get_opt(pools, Opts, []),
-    [start_pool(Host, Name, PoolOpts) || {Name, PoolOpts} <- PoolSpec],
+    ets:new(tab_name(), [set, public, named_table, {keypos, 2},
+                         {read_concurrency, true} | HeirOpt]),
+    [start_pool(Name, PoolOpts) || {Name, PoolOpts} <- Opts],
     ok.
 
--spec stop(ejabberd:server()) -> any().
-stop(Host) ->
-    stop_supervisor(Host),
-    ets:delete(tab_name(Host)).
-
 %%------------------------------------------------------------------------------
 %% internal functions
 
-make_pool(Host, Name, Opts) ->
+make_pool(Name, Opts) ->
     #pool{name = Name,
-          host = Host,
           http_host = gen_mod:get_opt(host, Opts, "http://localhost"),
```

Comment:
I guess the `http(s)://` prefix is required here. If so I'd rather call the option `URL` instead of host.  For me `host` refers only to the part of URL after the prefix and before first `/`.